### PR TITLE
optimize engine decorator

### PR DESCRIPTION
**Pull request by PPpro**
Re: 
https://github.com/cocos/3d-tasks/issues/9015
https://github.com/cocos/3d-tasks/issues/11934

### Changelog

* optimize engine decorator compiling, this optimization will
    * increase the instantiation performance
    * decrease the engine package size (from 4.36mb to 4.25 mb)

#### before
before

#### after
after

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by SantyWang — APPROVED** [review]

File: scripts/build-engine/src/optimize-decorators.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+export const fieldDecorators = [
+    "property",
+    // serializable
+    "serializable",
+    "formerlySerializedAs",
+    "editorOnly",
+    "uniquelyReferenced",
+    // eidtor
```
Comment (shrinktofit) [inline scripts/build-engine/src/optimize-decorators.ts]: mis-spelling

File: scripts/build-engine/src/optimize-decorators.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    "multiline",
+    "disallowAnimation",
+    // component
+    "requireComponent",
+    "executionOrder",
+    "disallowMultiple",
+    "allowReplicated",
+    // ccclass
+    "ccclass",
+];
```
Comment (shrinktofit) [inline scripts/build-engine/src/optimize-decorators.ts]: I wonder if this should be a config so developers can easialy add new decorators.

**Comment by shrinktofit** [comment]
Could you please add some basic unit tests?

**Comment by PPpro** [comment]
the result of auto test case is known issues

**Review by shrinktofit — APPROVED** [review]

**Comment by PPpro** [comment]
补充一下具体的测试数据：
image
表格文件：[装饰器优化性能测试.xls](https://github.com/cocos/cocos-engine/files/10191237/default.xls)
测试例：https://github.com/PPpro/performance-particle-test

数据说明：
- 性能数据为平均每实例化一个 prefab 的时间消耗
- 3.7 在加入装饰器优化之前，有一个更大幅度的优化，来自于 https://github.com/cocos/cocos-engine/pull/13638
- 包体方面从结果上看还是有小幅度上涨，装饰器优化降低了 0.11 mb

**Pull request merged**